templates/helm/templates/deployment.yaml

```diff
@@ -73,6 +73,11 @@ spec:
           value: {{ .Values.log.level | quote }}
         - name: ACK_RESOURCE_TAGS
           value: {{ join "," .Values.resourceTags | quote }}
+        {{- if .Values.deployment.secretEnvVars }}
+        envFrom:
+        - secretRef:
+            name: {{ include "app.fullname" . }}
```

Comment:
Consider potentially having a separate variable storing the name of the Secret to use (instead of hard-coding to a Secret named the same as the application)